### PR TITLE
refactor blockstores

### PR DESCRIPTION
**Pull request by raulk**
- [X] Place all blockstores into a single flattened package. Subpackages are only created for blockstores that depend on third-party components (e.g. badger).
- [X] Make the `blockstore.Blockstore` interface the union of the basic IPFS blockstore + Viewer, for cleanness and to remove type assertions. This interface will be enhanced soon with a `Sync()` method: 5057.
- [X] Introduce a `blockstore.Adapt(BasicBlockstore)` method to adapt a basic IPFS blockstore into a Lotus blockstore.
- [x] Rename blockstores for consistency.
- [ ] ~Add godocs to all blockstores.~ Still TODO.

---

Once we merge this, we'll be in a better position to add support for the `Sync()` method (#5057), and to continue extending the blockstore with specialised operations for better efficiency.

---

Partly addresses #5455.

### CODE REVIEW

**Comment by raulk** [comment]
This depends on #5425.

File: blockstore/blockstore.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	return callback(blk.RawData())
+}
+
+// Adapt adapts a standard blockstore to a Lotus blockstore by
+// enriching it with the extra methods that Lotus requires (e.g. View, Sync).
+//
+// View proxies over to Get and calls the callback with the value supplied by Get.
+// Sync noops.
+func Adapt(bs blockstore.Blockstore) Blockstore {
+	return &adaptedBlockstore{bs}
```
Comment (magik6k) [inline blockstore/blockstore.go:62]: Could try to cast `bs` to `Blockstore` and return that if successful

**Review by magik6k — APPROVED** [review]
Just 2 minor nits

**Comment by raulk** [comment]
@magik6k should be ready to put on a boat. @vyzo and I are preparing a bunch of PRs for the splitstore on top of this.

**Comment by raulk** [comment]
The build for lotus-soup failing is a red herring -- it's a flaw in the way that the CI jobs are set up. It will build correctly once this is merged into master.

**Review by magik6k — APPROVED** [review]

**Pull request merged**